chain/store/name.go

```diff
@@ -1,57 +1,127 @@
 package store
 
 import (
+	"bytes"
+	"io"
 	"strings"
+
+	"github.com/nknorg/nkn/common/serialization"
 )
 
+type nameInfo struct {
+	registrant []byte
+	expiresAt  uint32
+}
+
+func (ni *nameInfo) Serialize(w io.Writer) error {
+	if err := serialization.WriteVarBytes(w, ni.registrant); err != nil {
+		return err
+	}
+	if err := serialization.WriteUint32(w, ni.expiresAt); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (ni *nameInfo) Deserialize(r io.Reader) error {
+	var err error
+	ni.registrant, err = serialization.ReadVarBytes(r)
+	if err != nil {
+		return err
+	}
+	ni.expiresAt, err = serialization.ReadUint32(r)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (ni *nameInfo) Empty() bool {
+	return len(ni.registrant) == 0 && ni.expiresAt == 0
+}
+
 func getRegistrantId(registrant []byte) string {
 	return string(registrant)
 }
 
-func getNameId(name string) string {
-	return strings.ToLower(name)
+func getNameId(name string) []byte {
+	return []byte(strings.ToLower(name))
 }
 
-func (sdb *StateDB) updateName(registrant []byte, name string) error {
-	registrantId := getRegistrantId(registrant)
+func (sdb *StateDB) getNameInfo(name string) (*nameInfo, error) {
 	nameId := getNameId(name)
+	ni := new(nameInfo)
+
+	if v, ok := sdb.names.Load(nameId); ok {
+		if ni, ok := v.(*nameInfo); ok {
+			return ni, nil
+		}
+	}
 
-	err := sdb.trie.TryUpdate(append(NamePrefix, nameId...), registrant)
+	enc, err := sdb.trie.TryGet(append(NamePrefix, nameId...))
 	if err != nil {
-		return err
+		return nil, err
+	}
+	if len(enc) > 0 {
+		buff := bytes.NewBuffer(nil)
```

Comment:
Should be `NewBuffer(enc)` instead of `NewBuffer(nil)`

Reply:
done